client/post-editor/editor-media-advanced/index.jsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * External depencencies
+ */
+import React, { PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { toggleEditorMediaAdvanced } from 'state/ui/editor/media/actions';
+import Dialog from 'components/dialog';
+
+function EditorMediaAdvanced( { visible, toggleVisible } ) {
+	return (
+		<Dialog isVisible={ visible } onClose={ toggleVisible }>
+			WIP
+		</Dialog>
+	);
+}
+
+EditorMediaAdvanced.propTypes = {
+	visible: PropTypes.bool,
+	toggleVisible: PropTypes.func
+};
+
+EditorMediaAdvanced.defaultProps = {
+	visible: false,
+	toggleVisible: () => {}
+};
+
+export default connect( ( state ) => {
+	return {
+		visible: state.ui.editor.media.advanced
+	};
+}, ( dispatch ) => {
+	return bindActionCreators( {
+		toggleVisible: toggleEditorMediaAdvanced
+	}, dispatch );
+} )( EditorMediaAdvanced );
```

Comment:
Though I suppose there's also nothing stopping me from using line breaks with the arrow functions sans `return`:

```jsx
export default connect(
	state => ( { 
		visible: state.ui.editor.media.advanced
	} ),
	dispatch => bindActionCreators( { 
		toggleVisible: toggleEditorMediaAdvanced 
	}, dispatch )
)( EditorMediaAdvanced );
```

I like arrow functions, but I find I usually need to pause to really digest what's happening, e.g. that `=> ( {` has a special meaning compared to `=> {`. Maybe this is still the new-ness of ES6, but I'm not as quick to drop the old keywords if it helps with clarity.